pkg/ccl/backupccl/backup_destination.go

```diff
@@ -188,43 +191,72 @@ func resolveDest(
 // getBackupManifests fetches the backup manifest from a list of backup URIs.
 func getBackupManifests(
 	ctx context.Context,
+	mem *mon.BoundAccount,
 	user security.SQLUsername,
 	makeCloudStorage cloud.ExternalStorageFromURIFactory,
 	backupURIs []string,
 	encryption *jobspb.BackupEncryptionOptions,
-) ([]BackupManifest, error) {
+) ([]BackupManifest, int64, error) {
 	manifests := make([]BackupManifest, len(backupURIs))
 	if len(backupURIs) == 0 {
-		return manifests, nil
+		return manifests, 0, nil
 	}
 
+	memMu := struct {
+		syncutil.Mutex
+		total int64
+		mem   *mon.BoundAccount
+	}{}
+	memMu.mem = mem
+
 	g := ctxgroup.WithContext(ctx)
 	for i := range backupURIs {
 		i := i
+		// boundAccount isn't threadsafe so we'll make a new one for each goroutine
+		// to use while reading. When it is done, we'll shink and close that local
+		// one,
+		// local one but
+		// before reserving the size of from the passed one to account for the size
+		// of the returned item.
+		subMem := mem.Monitor().MakeBoundAccount()
 		g.GoCtx(func(ctx context.Context) error {
+			defer subMem.Close(ctx)
 			// TODO(lucy): We may want to upgrade the table descs to the newer
 			// foreign key representation here, in case there are backups from an
 			// older cluster. Keeping the descriptors as they are works for now
 			// since all we need to do is get the past backups' table/index spans,
 			// but it will be safer for future code to avoid having older-style
 			// descriptors around.
 			uri := backupURIs[i]
-			desc, err := ReadBackupManifestFromURI(
-				ctx, uri, user, makeCloudStorage, encryption,
+			desc, size, err := ReadBackupManifestFromURI(
+				ctx, &subMem, uri, user, makeCloudStorage, encryption,
 			)
 			if err != nil {
 				return errors.Wrapf(err, "failed to read backup from %q",
 					RedactURIForErrorMessage(uri))
 			}
 			manifests[i] = desc
-			return nil
+
+			// We'll lock the main one and reserve the indicicated size which is still
+			// currently reserved in subMem, so we'll briefly be double reserved but
+			// that's safer than releasing first.
+			memMu.Lock()
+			err = memMu.mem.Grow(ctx, size)
+			if err != nil {
+				memMu.total += size
```

Comment:
we're only incrementing memMu.total in the error condition, i might be missing something but shouldn't we being doing this unconditionally.? We return this field at the end of the method so that the caller can shrink the bound account once it stops holding on to the manifests.